
From: Maneesh Soni <maneesh@in.ibm.com>

Fix sysfs_rename_dir().  The sysfs_lookup() does not hash negative dentries
so just hash it before calling d_move.


---

 25-akpm/fs/sysfs/dir.c |    1 +
 1 files changed, 1 insertion(+)

diff -puN fs/sysfs/dir.c~sysfs-backing-store-sysfs_rename_dir-fix fs/sysfs/dir.c
--- 25/fs/sysfs/dir.c~sysfs-backing-store-sysfs_rename_dir-fix	2004-05-11 03:14:13.649908024 -0700
+++ 25-akpm/fs/sysfs/dir.c	2004-05-11 03:14:13.652907568 -0700
@@ -179,6 +179,7 @@ void sysfs_rename_dir(struct kobject * k
 	new_dentry = sysfs_get_dentry(parent, new_name);
 	if (!IS_ERR(new_dentry)) {
 		if (!new_dentry->d_inode) {
+			d_add(new_dentry, NULL);
 			d_move(kobj->dentry, new_dentry);
 			kobject_set_name(kobj,new_name);
 		}

_
